oak/server/BUILD

```diff
@@ -142,55 +133,29 @@ cc_library(
     deps = [],
 )
 
-cc_library(
-    name = "enclave_server",
-    srcs = [
-        "enclave_server.cc",
-    ],
-    hdrs = [
-        "enclave_server.h",
-    ],
+enclave_loader(
```

Comment:
Follow-up after seeing the rest of the PR: I think the current structure is a bit confusing, the same `oak_main.cc` is used in two rules, and different includes are added based on the actual target. Personally I would prefer separate mains and BUILD targets, especially since they may expose different flags and configuration options. In fact, that's already the case: the `enclave_path` flag does really apply to the dev target, right? And now I understand why you left the `path` constructor argument in `OakManager` even though it is unused, and you called both classes with the same name (all of which I think are not ideal choices).

Reply:
@tiziano88 yeah, the path is not great. We can have two mains, which is still not great as it will duplicate some code.

I don't really get the separate BUILD targets argument... we already have that. It's server:oak and server:oak_dev.

Reply:
Done